### PR TITLE
[CRT] Introduce corecrt.h

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Include this instead of crtdefs.h.
This is for compatibility with MS headers.

### CODE REVIEW

File: sdk/include/crt/corecrt.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#ifndef _CRTRESTRICT
+#define _CRTRESTRICT
+#endif
+
+#ifndef DEFINED_localeinfo_struct
+typedef struct localeinfo_struct {
+    pthreadlocinfo locinfo;
+    pthreadmbcinfo mbcinfo;
+}_locale_tstruct,*_locale_t;
```
Comment (HBelusca) [inline sdk/include/crt/corecrt.h:18]:
```suggestion
} _locale_tstruct, *_locale_t;
```

**Review by HBelusca — COMMENTED** [review]
It's compatibility with the CRT of which VS version? 2017, 2019, 2022?

**Comment by tkreuzer** [comment]
> It's compatibility with the CRT of which VS version? 2017, 2019, 2022?

Originally Visual Studio 2015. But these days the CRT is no longer part of VS, but part of the WIndows SDK. There it was introduced with the first Windows 10 SDK in 2015.
Both wine and mingw use this since 2019.

**Review by DarkFire01 — APPROVED** [review]

File: sdk/include/crt/yvals.h [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 #include <_mingw.h>
 /* TODO, don't include crtdef.h.  */
```
[the remark this answers is not in the document]
Comment (tkreuzer) [reply sdk/include/crt/yvals.h]: Removed the include completely.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**